lib/job/arXmlExportJob.php

```diff
@@ -0,0 +1,190 @@
+    $success = $this->createZipForDownload($tempPath);
+
+    if ($success !== true)
+    {
+      $this->error($this->i18n->__('Failed to create ZIP file.'));
+
+      return false;
+    }
+
+    // Mark job as complete and set download path
+    $this->info($this->i18n->__('Export and archiving complete.'));
+    $this->job->setStatusCompleted();
+    $this->job->downloadPath = $this->getDownloadRelativeFilePath();
+    $this->job->save();
+
+    return true;
+  }
+
+  /**
+   * Export search results as CSV
+   *
+   * @param string  Path of file to write CSV data to
+   *
+   * @return null
+   */
+  protected function exportResults($path, $params)
+  {
+    $exitCode= 0;
+    $resultIds = array();
+    $resultSet = QubitSearch::getInstance()->index->getType('QubitInformationObject')->search($this->search->getQuery(false, false));
+
+    foreach ($resultSet as $hit)
+    {
+      $resultIds[] = $hit->getId();
+    }
+
+    // If ElasticSearch document is stale (corresponding MySQL data deleted), ignore
+    if (empty($resultIds))
+    {
+      $this->error($this->i18n->__('Records not found in search index.'));
+      return;
+    }
+
+    foreach ($resultIds as $result)
+    {
+      $taskClassName = 'export:bulk';
+
+      $criteria = sprintf('--criteria="i.id = (%s)"', $result);
+      $currentLevelOnly = (true == $params['current-level-only']) ? '--current-level-only' : '';
+      $public = (true == $params['public']) ? '--public' : '';
+
+      $command = sprintf('php %s %s %s %s %s %s',
+        escapeshellarg(sfConfig::get('sf_root_dir').DIRECTORY_SEPARATOR.'symfony'),
+        escapeshellarg($taskClassName),
+        $criteria,
+        $currentLevelOnly,
+        $public,
+        escapeshellarg($path));
+
+      // Log the command string in the job output window.
+      $output[] = $command;
+
+      // stderr to stdout
+      $command .= ' 2>&1';
+
+      // Run
+      exec($command, $output, $exitCode);
+
+      // Throw exception if exit code is greater than zero
+      if (0 < $exitCode)
+      {
+        $output = implode($output, "<br />");
+
+        throw new sfException($output);
+      }
+      else
+      {
+        // Warnings
+        $this->errors = $output;
+      }
+    }
+  }
+
+  /**
+   * Create ZIP file from results
+   *
+   * @param string  Path of file to write CSV data to
+   *
+   * @return int  success bool
+   */
+  protected function createZipForDownload($path)
+  {
+    if (!is_writable($this->getJobsDownloadDirectory()))
+    {
+      return false;
+    }
+
+    $zip = new ZipArchive();
+
+    $success = $zip->open($this->getDownloadFilePath(), ZipArchive::CREATE | ZipArchive::OVERWRITE);
+
+    if ($success == true)
+    {
+      foreach(scandir($path) as $file)
+      {
+        if (!is_dir($file))
+        {
+          $zip->addFile($path . DIRECTORY_SEPARATOR . $file, $file);
+        }
+      }
+
+      $zip->close();
+    }
+
+    return $success;
+  }
```

Comment:
The createZipForDownload function could be moved to arBaseJob and removed from both CSV and XML export jobs